payments/tests/test_payments_models.py

```diff
@@ -0,0 +1,105 @@
+import random
+from decimal import Decimal
+
+import pytest
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError
+
+from payments.enums import AdditionalProductType, PeriodType, PriceUnits, ServiceType
+from payments.models import BerthPriceGroup, BerthProduct, DEFAULT_TAX_PERCENTAGE
+from payments.tests.factories import (
+    AdditionalProductFactory,
+    BerthProductFactory,
+    WinterStorageProductFactory,
+)
+from resources.models import BerthType
+
+
+def test_berth_product_invalid_price_unit():
+    with pytest.raises(ValidationError) as exception:
+        BerthProductFactory(price_unit=PriceUnits.PERCENTAGE)
+
+    errors = str(exception.value)
+    assert "percentage" in errors
+    assert "is not a valid choice" in errors
+
+
+def test_berth_product_invalid_tax():
+    with pytest.raises(ValidationError) as exception:
+        BerthProductFactory(tax_percentage=Decimal("10.00"))
+
+    errors = str(exception.value)
+    assert "tax_percentage" in errors
+    assert "is not a valid choice" in errors
+
+
+def test_berth_price_group_delete_berth_type_cascade(berth_price_group):
+    berth_price_group.berth_types.first().delete()
+
+    with pytest.raises(BerthProduct.DoesNotExist):
+        BerthProduct.objects.get(id=berth_price_group.id)
+
+
+def test_berth_product_delete_berth_type_product_group_is_none(berth_price_group):
+    berth_type = berth_price_group.berth_types.first()
+    BerthPriceGroup.objects.get(id=berth_price_group.id).delete()
+
+    assert BerthType.objects.get(id=berth_type.id).price_group is None
```

Comment:
are you testing that nullable FK with `SET_NULL` works properly? I think Django already covers a test like this 🙂

Reply:
Yup 😅 double checking that it's actually deleted, but I can remove it if it's redundant. :)

Reply:
I think we can trust Django on this one

Reply:
Alright! In Django we trust. 🙏